traits/src/fee.rs

```diff
@@ -0,0 +1,28 @@
+use frame_support::sp_runtime::{DispatchError, DispatchResult};
+
+//TODO: give better name and add doc comment
+pub trait InspectSufficiency<AssetId> {
+	fn is_sufficient(asset: AssetId) -> bool;
+
+	fn is_trade_supported(from: AssetId, into: AssetId) -> bool;
+}
+
+//TODO: give better name when we decicded if we wanna hardcode DOT or not
+pub trait InsufficientAssetTrader<AccountId, AssetId, Amount>: InspectSufficiency<AssetId> {
+	fn buy(
+		origin: &AccountId,
+		dest: &AccountId,
+		from: AssetId,
+		into: AssetId,
+		amount: Amount,
+		max_limit: Amount,
+	) -> DispatchResult;
+
+	fn pool_trade_fee(swap_amount: Amount) -> Result<Amount, DispatchError>;
+
+	fn get_amount_in_for_out(
```

Comment:
so it is consistent with other naming. 
```suggestion
	fn calculate_in_given_out(
```